desk/app/greeting.hoon

```diff
@@ -0,0 +1,157 @@
+/-  reel, chat
+/+  default-agent, verb, dbug, server
+::
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-0
+  ==
++$  state-0
+  $:  %0
+      token=cord
+      pals-tag=cord
+      dm-text=cord
+  ==
+--
+|%
+++  landing-page
+  |=  [token=cord dm-text=cord pals-tag=cord description=cord]
+  ^-  manx
+  ;html
+    ;head
+      ;title:"Greetings"
+    ==
+    ;body
+      ;form(method "post")
+        invite link token (should be URL-safe)
+        ;input(type "text", name "token", value "{(trip token)}");
+
+        pals tag for invitee
+        ;input(type "text", name "pals-tag", value "{(trip pals-tag)}");
+
+        direct message contents (sent to the invitee after clicking)
+        ;textarea(name "dm-text"):"{(trip dm-text)}"
+
+        invite link description
+        ;textarea(name "description"):"{(trip description)}"
+
+        On save, you\'ll get an invite link
+        ;button(type "submit"):"Save"
+      ==
+    ==
+  ==
+::
+++  sent-page
+  |=  [our=ship bait=cord token=cord]
+  ^-  manx
+  ;html
+    ;head
+      ;title:"Greetings"
+    ==
+    ;body
+      Here is your invite link:
+
+      {(trip bait)}{<our>}/{(trip token)}
+    ==
+  ==
+++  frisk  ::  parse url-encoded form args
+  |=  body=@t
+  %-  ~(gas by *(map @t @t))
+  (fall (rush body yquy:de-purl:html) ~)
+--
+::
+=|  state-0
+=*  state  -
+%-  agent:dbug
+^-  agent:gall
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %.n) bowl)
+::
+++  on-init
+  :_  this
+  :~  [%pass /bite-wire %agent [our.bowl %reel] %watch /bites]
+      [%pass /eyre/connect %arvo %e %connect [~ /greeting] dap.bowl]
+  ==
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  ?+  mark  !!
+    %leave  :_  this  ~[[%pass /bite-wire %agent [our.bowl %reel] %leave ~]]
+    %watch  :_  this  ~[[%pass /bite-wire %agent [our.bowl %reel] %watch /bites]]
+    ::
+      %handle-http-request
+    =+  !<([id=@ta inbound-request:eyre] vase)
+    |^
+    =/  line=request-line:server  (parse-request-line:server url.request)
+    ?+    method.request  :_  this  (give not-found:gen:server)
+        %'GET'
+      =/  description
+        .^(cord %gx [(scot %p our.bowl) %reel (scot %da now.bowl) %description token %noun ~])
+      :_  this  (give (manx-response:gen:server (landing-page token dm-text pals-tag description)))
+        %'POST'
+      ?~  body.request
+        :_  this  (give not-found:gen:server)
+      =/  params  ~(got by (frisk q.u.body.request))
+      =/  bait
+        .^(cord %gx [(scot %p our.bowl) %reel (scot %da now.bowl) %service %noun ~])
+      :_  this(token (params 'token'), dm-text (params 'dm-text'), pals-tag (params 'pals-tag'))
+      :-  [%pass /describe %agent [our.bowl %reel] %poke %reel-describe !>([(params 'token') (params 'description')])]
+      (give (manx-response:gen:server (sent-page our.bowl bait token)))
```

Comment:
`token` hasn't been updated yet here because it's happening in the parallel `:_` branch, so it doesn't show up in the sent page.

![Screenshot from 2023-01-27 16-56-01](https://user-images.githubusercontent.com/5553176/215227325-61c30c65-3a69-4180-9543-0fff630bdbc8.png)